src/features/movies/MovieDetails/index.js

```diff
@@ -16,73 +19,104 @@ import {
   AdditionalInfoWrapper,
 } from "../../../common/Details";
 import { PosterWrapper, PosterDisplay, MainInfoWrapper, MainInfoTitle } from "../../movies/MovieDetails/styled";
+import Loading from "../../../common/Loading";
+import Error from "../../../common/Error";
 import PeopleList from "../../people/PeopleList";
-import image from "../../../common/images/poster.png";
+import { changeMovieId, fetchMovieDetails, selectError, selectLoading, selectMovieDetails } from "./movieDetailsSlice";
+import { apiBackdrop } from "../../../apiData";
 
 const MovieDetails = () => {
+  const dispatch = useDispatch();
+  const { id } = useParams();
+  const loading = useSelector(selectLoading);
+  const error = useSelector(selectError);
+  const movie = useSelector(selectMovieDetails);
+  const releaseDate = movie.release_date;
+  const [date, setDate] = useState("");
+
+  useEffect(() => {
+    dispatch(changeMovieId(id));
+    dispatch(fetchMovieDetails());
+  }, []);
+
+  useEffect(() => {
+    if (releaseDate) {
+      const [year, month, day] = releaseDate.split("-");
+      setDate(day + "." + month + "." + year);
+    }
+  }, [releaseDate]);
+  
+  if (loading) {
+    return <Loading />;
+  } else if (error) {
+    return <Error />;
+  }
+
   return (
     <>
       <PosterDisplay>
-        <PosterWrapper>
+        <PosterWrapper style={{ backgroundImage: `url(${apiBackdrop}${movie.backdrop_path})` }}>
```

Comment:
Maybe we should add MovieBackdrop as div and implemented this backdrop somethink like this:
`{movie.backdrop_path && (
        <MovieBackdrop
          src={{${apiBackdrop}${movie.backdrop_path}}
        >`

Reply:
Thanks for suggestion. I moved Backrop to separate component and implemented ternary to check if there is backdrop image in fetched data.